src/oss/langchain-messages.mdx

```diff
@@ -2,14 +2,488 @@
+
+# Basic instruction
+system_msg = SystemMessage("You are a helpful coding assistant.")
+
+# Detailed persona
+system_msg = SystemMessage("""
+You are a senior Python developer with expertise in web frameworks.
+Always provide code examples and explain your reasoning.
+Be concise but thorough in your explanations.
+""")
+
+# Use with a query
+messages = [
+    system_msg,
+    HumanMessage("How do I create a REST API?")
+]
+response = model.invoke(messages)
+```
+
+### Provider Handling
+
+Different providers handle system messages in various ways:
+
+- **Native support**: Message included with `system` role
+- **API parameter**: Content passed as separate system parameter  
+- **No support**: Content merged into first human message
+
+LangChain handles these differences automatically.
+
+### Advanced System Message Patterns
+
+TODO keep? or link out to advanced guide?
+
+```python
+# Multi-part system instructions
+system_parts = [
+    "You are an expert data analyst.",
+    "Always include confidence levels in your analysis.",
+    "Provide actionable insights based on the data."
+]
+system_msg = SystemMessage("\n".join(system_parts))
+
+# Dynamic system messages
+def create_system_message(domain: str, style: str) -> SystemMessage:
+    return SystemMessage(f"You are a {domain} expert. Respond in a {style} style.")
+
+system_msg = create_system_message("finance", "professional")
+```
+
 ## Human Message
 
+Human messages represent user input and interactions. They can contain text, images, audio, files, and any other multimodal content.
+
+### Text Content
+
+```python
+from langchain_core.messages import HumanMessage
+
+human_msg = HumanMessage("What is machine learning?")
+
+# Use with models
+# TODO which pattern to show here? or all?
+response = model.invoke([human_msg])
+response = model.invoke(human_msg)
+response = model.invoke("What is machine learning?")
+```
+
+### String Shortcut
+
+TODO keep??
+
+For quick testing, strings are automatically converted to HumanMessage:
+
+```python
+# These are equivalent
+response1 = model.invoke([HumanMessage("Hello")])
+response2 = model.invoke("Hello")  # Automatically converted
+```
+
+### Message Metadata
+
+```python
+# Add metadata
+human_msg = HumanMessage(
+    content="Hello!",
+    name="alice",  # Identify different users
+    id="msg_123"   # Unique identifier for tracing
+)
+
+# Use in multi-user scenarios
+messages = [
+    HumanMessage("I like apples", name="alice"),
+    HumanMessage("I prefer oranges", name="bob"),
+    HumanMessage("What fruits do Alice and Bob like?", name="charlie")
+]
+```
+
+TODO make sure how `name` field is used across different providers is clear
+
 ## AI Message
 
+AI messages represent model responses and contain the output of chat model invocations. They can include text, tool calls, and rich metadata.
+
+### Basic AI Messages
+
+```python
+from langchain_core.messages import AIMessage
+
+# Simple response
```

Comment:
this is confusing snippet. i think can just show model generating one, and then show that response is that type. if we want to show how to construct an AI message by hand, can be separate section and should be movitated (wanting to prompt with a lsit of messages)